x-pack/plugins/security_solution/public/common/components/user_privileges/use_endpoint_privileges.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useRef, useState } from 'react';
+import { useCurrentUser, useHttp } from '../../lib/kibana';
+import { appRoutesService, CheckPermissionsResponse } from '../../../../../fleet/common';
+
+export interface EndpointPrivileges {
+  loading: boolean;
+  /** If user has permissions to access Fleet */
+  canAccessFleet: boolean;
+  /** If user has permissions to access Endpoint management (includes check to ensure they also have access to fleet) */
+  canAccessEndpointManagement: boolean;
+}
+
+/**
+ * Retrieve the endpoint privileges for the current user.
+ *
+ * **NOTE:** Consider using `usePrivileges().endpointPrivileges` instead of this hook in order
+ * to keep API calls to a minimum.
+ */
+export const useEndpointPrivileges = (): EndpointPrivileges => {
+  const http = useHttp();
+  const user = useCurrentUser();
+  const isMounted = useRef<boolean>(true);
+  const [privileges, setPrivileges] = useState<EndpointPrivileges>({
+    loading: true,
+    canAccessFleet: false,
+    canAccessEndpointManagement: false,
+  });
+  const [canAccessFleet, setCanAccessFleet] = useState<boolean>(false);
+  const [isSuperUser, setIsSuperUser] = useState<boolean>(false);
+  const [fleetCheckDone, setFleetCheckDone] = useState<boolean>(false);
+  const [userRoleCheckDone, setUserRoleCheckDone] = useState<boolean>(false);
+
+  // Check if user can access fleet
+  useEffect(() => {
+    (async () => {
+      http
+        .get<CheckPermissionsResponse>(appRoutesService.getCheckPermissionsPath())
```

Comment:
Do we need to show an error message to the user if the HTTP request fails? You can use `useAppToasts` for that.

```ts
const { addError } = useAppToasts();
// And then
catch(error => addError(error, { title: i18n.ENDPOINT_PRIVILEGES_FETCH_FAILURE }))
```

Reply:
I'm always hesitant to show UI feedback from an area of the code that has no control over the user experience and like for that to be delegated to the UI code that is currently in front of the user. Thinking about this here, I guess (but probably highly unlikely) that there could be a case where Fleet plugin is disabled, in which case the plugin as a whole would not be present and API calls would fail. In that case we likely don't want toast messages to be shown. There may be other cases as well.

If you are ok with it, let's leave it as is for now. we can adjust if needed in the future.

Reply:
Fwiw, we show an error toast when `listPrivileges` or `detectionEnginePrivileges` request fails. But if you feel strongly about not showing errors when we fetch `endpointPrivileges`, I'm okay with that.